src/Maker/MakeResetPassword.php

```diff
@@ -194,23 +203,84 @@ public function generate(InputInterface $input, ConsoleStyle $io, Generator $gen
             ]
         );
 
-        $generator->generateClass(
-            $requestClassNameDetails->getFullName(),
-            'resetPassword/ResetPasswordRequest.tpl.php',
-            [
-                'repository_class_name' => $repositoryClassNameDetails->getFullName(),
-                'user_full_class_name' => $userClassNameDetails->getFullName(),
-            ]
+        $entityClassGenerator = new EntityClassGenerator($generator, $this->doctrineHelper);
+
+        $pathRequestEntity = $entityClassGenerator->generateEntityClass($requestClassNameDetails, false);
+
+        $generator->writeChanges();
+
+        $manipulator = new ClassSourceManipulator(
+            $this->fileManager->getFileContents($pathRequestEntity)
         );
 
-        $generator->generateClass(
-            $repositoryClassNameDetails->getFullName(),
-            'resetPassword/ResetPasswordRequestRepository.tpl.php',
-            [
-                'request_class_full_name' => $requestClassNameDetails->getFullName(),
-                'request_class_name' => $requestClassNameDetails->getShortName(),
-            ]
+        $manipulator->addInterface(ResetPasswordRequestInterface::class);
+
+        $manipulator->clearClassNodeStmts();
+
+        $manipulator->addTrait(ResetPasswordRequestTrait::class);
+
+        $manipulator->addConstructor([
+                ['user', null, 'object'],
+                ['expiresAt', null, '\DateTimeInterface'],
+                ['selector', null, 'string'],
+                ['hashedToken', null, 'string'],
+            ], <<<'CODE'
+<?php
+$this->user = $user;
+$this->initialize($expiresAt, $selector, $hashedToken);
+CODE
+        );
+        $manipulator->addProperty('id', [
+            '@ORM\Id()',
+            '@ORM\GeneratedValue()',
+            '@ORM\Column(type="integer")',
+        ]);
+
+        $manipulator->addProperty('user', [
+            '@ORM\ManyToOne(targetEntity="App\Entity\User")',
+        ]);
+
+        $manipulator->addGetter('user', 'object', false);
```

Comment:
Could we use `$manipulator->addSingularRelation()` ? Maybe with a new option to avoid the setter?

Reply:
addSingularRelation is a private method wich is indirectly called. It would make the methods less "clean" and I don't think that's necessary, maybe we can look at this later if we have other cases ? but if you insist I can looking for .

Reply:
The problem is that we need to centralize the `@ORM\ManyToOne` code - I don't want that duplicated. In fact, recently, I merged a PR that updates the relationship methods on `ClassSourceManipulator` so that they generate like this:

```
@ORM\ManyToOne(targetEntity=User::class)
```

By centralizing the logic, we'll keep the way we generate things consistent. I didn't realize that `addSingularRelation` was private. So yes, we will need to make this public or think of a new public method we could add that would call (it just depends on what's cleanest - I'm not looking deeply at the code at the moment) :)

Reply:
If you are okay we can create the relation with the setter. Is it a problem ? Usually it's the classic behavior when maker creates the relations.

Reply:
I can add this in BaseRelation

Reply:
`

    public function isAvoidSetter() {
        return $this->avoidSetter;
    }

    public function avoidSetter(bool $avoidSetter = true) {
        $this->avoidSetter = $avoidSetter;

        return $this;
    }`

Reply:
And avoid the setter in addSingularRelation. I just tried and it works.

Reply:
In makeResetPassword
`

        $manipulator->addManyToOneRelation((new RelationManyToOne())
            ->setPropertyName('user')
            ->setTargetClassName("App\Entity\User")
            ->setIsNullable(true)
            ->setMapInverseRelation(false)
            ->avoidSetter()
        );`

Reply:
But i have a problem with the interface of the ResetPasswordRequestInterface
I need to type return with an object, by default it returns ?User

interface ResetPasswordRequestInterface
{
    /**
     * Get the user whom requested a password reset.
     */
    public function getUser(): object;

Reply:
https://github.com/SymfonyCasts/reset-password-bundle/blob/96391ea830a24bc2dcfc4437edd6648cc55ebcfd/src/Model/ResetPasswordRequestInterface.php#L38

Reply:
I can add this in BaseRelation
`

    public function getReturnType() {
        return $this->returnType;
    }

    public function setReturnType(string $returnType) {
        $this->returnType = $returnType;

        return $this;
    }`

Reply:
And tests works

Reply:
In makeResetPassword
`$manipulator->addManyToOneRelation((new RelationManyToOne()) ->setPropertyName('user') ->setTargetClassName("App\Entity\User") ->setIsNullable(true) ->setMapInverseRelation(false) ->setReturnType('object', false) ->avoidSetter() );`

Reply:
Let's do this!

Reply:
Done.